internal/layer2/announcer.go

```diff
@@ -131,24 +137,49 @@ func (a *Announce) updateInterfaces() {
 	return
 }
 
-func (a *Announce) spam(name string) {
-	// TODO: should abort if we lose control of the IP mid-spam.
-	start := time.Now()
-	for time.Since(start) < 5*time.Second {
-		if err := a.gratuitous(name); err != nil {
-			a.logger.Log("op", "gratuitousAnnounce", "error", err, "service", name, "msg", "failed to make gratuitous IP announcement")
+func (a *Announce) spamLoop() {
+	m := map[string]time.Time{}
+	// see https://github.com/metallb/metallb/issues/172 for the 1100 choice
+	ticker := time.NewTicker(1100 * time.Millisecond)
+	for {
+		select {
+		case ip := <-a.spamCh:
+			ips := ip.String()
```

Comment:
I find `ips` confusing as it sounds like the var holds multiple IP addresses.

I suggest doing the following:

```go
case e := <-a.spamCh:
    ip := e.String()
```

Reply:
I went with `ipStr` because I also use `ip` in `a.spam(ip)`